### PR TITLE
11 get article by topic

### PR DESCRIPTION
**Pull request by avr87**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -123,7 +123,46 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(1);
+        body.articles.forEach((article) => {
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:136]: We should hard code the topic property here as every article should have the topic 'cats' if we've queried it correctly.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -123,7 +123,46 @@ describe("GET /api/articles", () => {
...
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  test("GET 200 responds with an array of the articles with the requested topic.", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:148]: Are we testing for any different behaviours in this test? If it auto passed or both this and the previous test were written, both failed and then both passed it's an indicator of an unnecessary test 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -123,7 +123,46 @@ describe("GET /api/articles", () => {
...
+  test("GET 404 responds with an error message when there is no article available under this topic", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toEqual(
+          "There is currently no article with this topic available"
+        );
+      });
+  });
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:163]: Nice one, we may also want to consider a valid topic that exists but happens to have no articles associated with it.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -322,3 +361,12 @@ describe("GET: /api/users", () => {
...
 });
+
+// FEATURE REQUEST The endpoint should also accept the following query:
+
+// topic, which filters the articles by the topic value specified in the query. If the query is omitted, the endpoint should respond with all articles.
+// Consider what errors could occur with this endpoint, and make sure to test for them.
+
+// You should not have to amend any previous tests.
+
+// Remember to add a description of this endpoint to your /api endpoint.
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:372]: pls remove comments and console.logs before committing 👍

File: __tests__/app.test.js (hunk 3 above)
Comment (avr87) [reply __tests__/app.test.js:163]: @nc-richard-neat Thanks, for the feedback. This one tests for an existing topic that has no articles available.  Am I missing a different test?

**Pull request merged**